src/Facet/Core/Type.hs

```diff
@@ -69,57 +75,70 @@ metavar = var . Free . Left
 
 
 var :: Var (Either Meta Level) -> Type
-var v = VNe v Nil Nil
+var v = VNe v Nil
 
 
-unComp :: Has Empty sig m => Type -> m ([Type], Type)
+unComp :: Has Empty sig m => Type -> m ([Interface Type], Type)
 unComp = \case
   VComp sig _T -> pure (sig, _T)
   _T           -> empty
 
 
+data Subject
+  = SK Kind
+  | ST Type
+
+subjectType :: Subject -> Maybe Type
+subjectType = \case
+  SK _K -> empty
+  ST _T -> pure _T
```

Comment:
This is used by various errors and things. It was introduced during an experiment with polarization that I ended up backing out, but it’s still kind of handy and I might end up using it in contexts at some point in place of `Either`.